src/radical/pilot/compute_pilot.py

```diff
@@ -547,68 +541,19 @@ def cancel(self):
     # --------------------------------------------------------------------------
     #
     def stage_in(self, directives):
-        """Stages the content of the staging directive into the pilot's
-        staging area"""
-
-        # Wait until we can assume the pilot directory to be created
-        # # FIXME: can't we create it ourself?
-        if self.state == rps.NEW:
-            self.wait(state=[rps.PMGR_LAUNCHING_PENDING, rps.PMGR_LAUNCHING,
-                             rps.PMGR_ACTIVE_PENDING,    rps.PMGR_ACTIVE])
-        elif self.state in rps.FINAL:
-            raise Exception("Pilot already finished, no need to stage anymore!")
-
-        # NOTE: no unit sandboxes defined!
-        src_context = {'pwd'      : self._client_sandbox,     # !!!
-                       'pilot'    : self._pilot_sandbox,
-                       'resource' : self._resource_sandbox}
-        tgt_context = {'pwd'      : self._pilot_sandbox,      # !!!
-                       'pilot'    : self._pilot_sandbox,
-                       'resource' : self._resource_sandbox}
-
-        # Iterate over all directives
-        for sd in expand_staging_directives(directives):
-
-            # TODO: respect flags in directive
-
-            action = sd['action']
-            flags  = sd['flags']
-            did    = sd['uid']
-            src    = sd['source']
-            tgt    = sd['target']
-
-            assert(action in [COPY, LINK, MOVE, TRANSFER])
```

Comment:
This is something I wanted to talk to you about, `asserts` are good but we should not depend on them, you can run a python program with the Optimized flags (i.e. `-o` or `-oo`) all of these statements are ignored so we should not necessarily depend too much on them. (Just FYI~.)

https://docs.python.org/2.7/tutorial/modules.html#compiled-python-files

Reply:
(I know this line is being removed, which is fine, it is just that there are a lot of `assert` statements all over Pilot).

Reply:
yeah, I absolutely agree on that!  I only use assert on things which actually never should be incorrect - they should only get triggered if a protocol or call convention changes in one place but is not updated in another place, or stuff like that. Runtime checks should not be asserted otherwise...

Reply:
Let me correct: I _should_ only use asserts like the above :D  I'll do a round of checks to make sure that's the case...